test/integration_test.go

```diff
@@ -148,18 +148,24 @@ func (ts *IntegrationTestSuite) SetupTest() {
 	ts.seq++
 	ts.activities.clearInvoked()
 	ts.taskListName = fmt.Sprintf("tl-%v", ts.seq)
-	ts.worker = worker.New(ts.rpcClient.Interface, domainName, ts.taskListName, worker.Options{
-		DisableStickyExecution: ts.config.IsStickyOff,
-		Logger:                 zaptest.NewLogger(ts.T()),
-		ContextPropagators:     []workflow.ContextPropagator{NewStringMapPropagator([]string{testContextKey})},
-	})
 	ts.tracer = newtracingInterceptorFactory()
+}
+
+func (ts *IntegrationTestSuite) BeforeTest(suiteName, testName string) {
 	options := worker.Options{
 		DisableStickyExecution:            ts.config.IsStickyOff,
 		Logger:                            zaptest.NewLogger(ts.T()),
 		WorkflowInterceptorChainFactories: []interceptors.WorkflowInterceptorFactory{ts.tracer},
 		ContextPropagators:                []workflow.ContextPropagator{NewStringMapPropagator([]string{testContextKey})},
 	}
+
+	if testName == "TestNonDeterministicWorkflowQuery" || testName == "TestNonDeterministicWorkflowFailPolicy" {
```

Comment:
I completely believe that this is the easy option, but uh.  tbh this kind of thing is why I want to get rid of these `testify.Suite`s entirely.  spooky action at a distance and ad-hoc DSLs are a massive pain much of the time, if we didn't have this it'd have just been an argument on a `func setup(...)` and it would be obvious and controllable.

Reply:
This workaround demonstrates one of the weaknesses of test suites where your setup and teardown can only diverge based on test name whereas a proper table driven test would have a `worker_options` field to be specified per test case.
If I find some free time I may rewrite this file by ripping out suite and asserts in a follow up PR

Reply:
yea, possibly not worth it in this PR.  more of a "let's try hard to keep this to a minimum please" request - short-term it's fine.